src/scene/layer-composition.js

```diff
@@ -466,31 +317,132 @@ class LayerComposition extends EventHandler {
             }
 
             this._renderActions.length = renderActionCount;
+        }
 
+        if ((result & COMPUPDATED_LIGHTS) || (result & COMPUPDATED_CAMERAS)) {
             this._logRenderActions();
         }
 
-        var arr;
-        if ((result & COMPUPDATED_LIGHTS) || (result & COMPUPDATED_CAMERAS)) {
-            // cameras/lights changed
-            this._globalLightCameraIds.length = 0;
-            for (l = 0; l < this._globalLightCameras.length; l++) {
-                arr = [];
-                for (i = 0; i < this._globalLightCameras[l].length; i++) {
-                    index = this.cameras.indexOf( this._globalLightCameras[l][i] );
-                    if (index < 0) {
-                        // #ifdef DEBUG
-                        console.warn("Can't find _globalLightCameras[l][i] in cameras");
-                        // #endif
-                        continue;
+        return result;
+    }
+
+    updateShadowCasters() {
+
```

Comment:
I'm really not sure about this blank line policy.

I don't like the idea of the engine having inconsistent formatting - some code having (seemingly random) blank lines and other code not.

And I don't favour changing the entire engine to having blank lines.

Reply:
I like empty line before a comment line .. so this is what I did here. But perhaps we need to have a discussion on this.